### PR TITLE
Collected small changes and fixes

### PR DESCRIPTION
**Pull request by akohlmey**
**Summary**

This pull request combines multiple small changes and fixes.

**Related Issue(s)**

N/A

**Author(s)**

Axel Kohlmeyer, Temple U

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

N/A

**Implementation Notes**

The following individual changes are included:
- small documentation updates and corrections
- bugfix for SNAP support in ML-POD from Aidan
- bugfixes for building the LEPTON package with the traditional make build procedure
- avoid ambiguities for calling math functions. required to compile LAMMPS on OpenIndiana/Solaris
- update error messages in group command
- add more unit tests to the group command/class tester, update for changed error messages

**Post Submission Checklist**

- [x] The feature or features in this pull request is complete
- [x] Licensing information is complete
- [x] Corresponding author information is complete
- [x] The source code follows the LAMMPS formatting guidelines
- [x] Suitable new documentation files and/or updates to the existing docs are included
- [x] The added/updated documentation is integrated and tested with the documentation build system
- [x] The feature has been verified to work with the conventional build system
- [x] The feature has been verified to work with the CMake based build system
- [x] Suitable tests have been added to the unittest tree.

### CODE REVIEW

File: src/group.cpp [hunk 1]
```diff
@@ -155,7 +155,7 @@ void Group::assign(int narg, char **arg)
   bool created = false;
 
   if (igroup == -1) {
-    if (ngroup == MAX_GROUP) error->all(FLERR,"Too many groups");
+    if (ngroup == MAX_GROUP) error->all(FLERR,"Too many groups (max. {})", MAX_GROUP);
```
Comment (sjplimp) [inline src/group.cpp]: no period after max please

File: src/group.cpp [hunk 2]
```diff
@@ -569,7 +566,7 @@ void Group::create(const std::string &name, int *flag)
   int igroup = find(name);
 
   if (igroup == -1) {
-    if (ngroup == MAX_GROUP) error->all(FLERR,"Too many groups");
+    if (ngroup == MAX_GROUP) error->all(FLERR,"Too many groups (max. {})", MAX_GROUP);
```
Comment (sjplimp) [inline src/group.cpp]: ditto

File: src/group.cpp [hunk 3]
```diff
@@ -606,7 +603,7 @@ int Group::find_or_create(const char *name)
   int igroup = find(name);
   if (igroup >= 0) return igroup;
 
-  if (ngroup == MAX_GROUP) error->all(FLERR,"Too many groups");
+  if (ngroup == MAX_GROUP) error->all(FLERR,"Too many groups (max. {})", MAX_GROUP);
```
Comment (sjplimp) [inline src/group.cpp]: ditto

Comment (akohlmey) [reply src/group.cpp]: done

**Review by sjplimp — APPROVED** [review]

**Pull request merged**